tests/test_commonline_sync3n.py

```diff
@@ -136,3 +141,45 @@ def test_estimate_rotations(source_orientation_objs):
     if src.offsets.all() != 0:
         tol = 4
     mean_aligned_angular_distance(orient_est.rotations, src.rotations, degree_tol=tol)
+
+
+@pytest.mark.expensive
+def test_weighted_sync3n(source_orientation_objs):
+    """
+    Test alternative Sync3N configuration code paths.
+    """
+    src, _ = source_orientation_objs
+
+    # Search for common lines over less shifts for 0 offsets.
+    max_shift = 1 / src.L
+    shift_step = 1
+    if src.offsets.all() != 0:
+        max_shift = 0.20
+        shift_step = 0.25  # Reduce shift steps for non-integer offsets of Simulation.
+
+    orient_est = CLSync3N(
+        src,
+        max_shift=max_shift,
+        shift_step=shift_step,
+        seed=789,
+        S_weighting=True,
+        J_weighting=True,
+        full_width=2,
+        sigma=2.9,
+    )
+    # Estimate rotations
+    orient_est.estimate_rotations()
+
+    gt_clmatrix = rots_to_clmatrix(src.rotations, orient_est.n_theta)
+
+    angle_diffs = abs(orient_est.clmatrix - gt_clmatrix) * 360 / orient_est.n_theta
+
+    # Count number of estimates within 5 degrees of ground truth.
+    within_5 = np.sum((angle_diffs - 360) % 360 < 5)
```

Comment:
You mentioned this line in slack so just wanted to add a comment. After looking at it a little closer it appears to be doing what we expect.

Reply:
Yeah, I agree, but am still not sure why I was seeing so many 359 at the end... I think it might be related to the `-1` skip convention?.. tbd.